src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructureUnitTest.java

```diff
@@ -1,19 +1,359 @@
+            final List<String> sortedNodes = new ArrayList<>(Sets.union(getAncestors(nodeKey),
+                    Collections.singleton(nodeKey)));
+            sortedNodes.sort(Comparator.comparingInt(topologicalOrderMap::get));
+            return sortedNodes;
+        }
+
+        /**
+         * Brute-force method
+         */
+        List<String> getTopologicalOrderForNodeMutation(final String nodeKey) {
+            final Set<String> mutatedNodeAndDescendants = Sets.union(getDescendents(nodeKey),
+                    Collections.singleton(nodeKey));
+            final Set<String> ancestorsOfDescendants = getDescendents(nodeKey)
+                    .stream()
+                    .map(this::getAncestors)
+                    .flatMap(Set::stream)
+                    .collect(Collectors.toSet());
+            final Set<String> involvedNodes = Sets.union(mutatedNodeAndDescendants, ancestorsOfDescendants);
+            final List<String> topologicallySortedInvolvedNodes = new ArrayList<>(involvedNodes);
+            topologicallySortedInvolvedNodes.sort(Comparator.comparingInt(topologicalOrderMap::get));
+            return topologicallySortedInvolvedNodes;
+        }
+
+        /**
+         * Brute-force method
+         */
+        List<String> getTopologicalOrderForTagEvaluation(final String tag) {
+            final Set<String> taggedNodes = nodeKeysSet.stream()
+                    .filter(nodeKey -> getInducedTags(nodeKey).contains(tag))
+                    .collect(Collectors.toSet());
+            final Set<String> taggedNodesAndTheirAncestors = Sets.union(taggedNodes,
+                    taggedNodes.stream()
+                            .map(this::getAncestors)
+                            .flatMap(Set::stream)
+                            .collect(Collectors.toSet()));
+            final List<String> topologicallySortedtaggedNodesAndTheirAncestors = new ArrayList<>(taggedNodesAndTheirAncestors);
+            topologicallySortedtaggedNodesAndTheirAncestors.sort(Comparator.comparingInt(topologicalOrderMap::get));
+            return topologicallySortedtaggedNodesAndTheirAncestors;
+        }
+
+        Set<CacheNode> getEquivalentCacheNodeSet() {
+            final List<String> shuffledNodeList = new ArrayList<>(nodeKeysSet);
+            Collections.shuffle(shuffledNodeList, rng);
+            return shuffledNodeList.stream()
+                    .map(nodeKey -> topologicalOrderMap.get(nodeKey) == 0
+                            ? new PrimitiveCacheNode(nodeKey, tagsMap.get(nodeKey), null)
+                            : new ComputableCacheNode(nodeKey, tagsMap.get(nodeKey), getParents(nodeKey), null, true))
+                    .collect(Collectors.toSet());
+        }
+
+        private static Set<String> getRandomNodeKeys(final int maxNodes) {
+            return IntStream.range(0, rng.nextInt(maxNodes) + 1)
+                    .mapToObj(i -> "NODE_KEY_" + RandomStringUtils.randomAlphanumeric(NODE_KEY_LENGTH))
```

Comment:
I put a condition to ensure that all keys are unique. Previously, the probability of collision was about `1e-49` (very passive aggressive :D).